lib/rules/prefer-const.js

```diff
@@ -9,7 +9,7 @@
 // Helpers
 //------------------------------------------------------------------------------
 
-const PATTERN_TYPE = /^(?:.+?Pattern|RestElement|Property)$/;
+const PATTERN_TYPE = /^(?:.+?Pattern|RestElement|ExperimentalRestProperty|Property)$/;
```

Comment:
Given what we've uncovered in the discussion, this should probably include RestProperty as well.

If at all possible, a test which simulates the babel-eslint parser's output would also be awesome, but don't put too much effort into that.